File: coding/02nd December 2025/Divide Array Into Equal Pairs.java
//Divide Array Into Equal Pairs
class Solution 
{
    public boolean divideArray(int[] nums) 
    {
        HashMap<Integer, Integer> map = new HashMap<Integer, Integer>();
        for(int ele : nums)
        {
            if(map.containsKey(ele))
                map.put(ele, map.get(ele) + 1);
            else
                map.put(ele, 1);
        }
        for(int k : map.keySet())
            if(map.get(k) % 2 != 0)
                return false;
        return true;
    }
}

File: Coding in Python/14th February 2026/Divide Array Into Equal Pairs.py
#Divide Array Into Equal Pairs
class Solution:
    def divideArray(self, nums: List[int]) -> bool:
        dict1 = {}
        for ele in nums:
            if ele in dict1:
                dict1[ele] += 1
            else:
                dict1[ele] = 1
        for k in dict1:
            if dict1[k] % 2 != 0:
                return False
        return True
